scipy/optimize/_numdiff.py

```diff
@@ -206,6 +212,9 @@ def approx_derivative(fun, x0, method='3-point', rel_step=None, f0=None,
                      that the user function is real-valued and can be
                      analytically continued to the complex plane. Otherwise,
                      produces bogus results.
+
+        The method 'cs' is not available when `return_linear_operator`
```

Comment:
The basic idea of is to compute of this new functionality I am introducing is to be able to compute: ``jac.dot(p)`` cheaply. We can do it by computing: (2-point case)
``(v(x + h*p) - v(x))/h``
The idea of using a complex step is to use an complex direction in order to improve the accuracy. This however does not make much sense in the linear operator case since we already have a fixed direction ``p`.

Does it make sense for you?